airflow/api_fastapi/core_api/routes/public/job.py

```diff
@@ -0,0 +1,128 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from fastapi import Depends, status
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+from typing_extensions import Annotated
+
+from airflow.api_fastapi.common.db.common import (
+    get_session,
+    paginated_select,
+)
+from airflow.api_fastapi.common.parameters import (
+    QueryLimit,
+    QueryOffset,
+    RangeFilter,
+    SortParam,
+    datetime_range_filter_factory,
+)
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.datamodels.job import (
+    JobCollectionResponse,
+    JobResponse,
+)
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.jobs.job import Job
+from airflow.utils.state import JobState
+
+job_router = AirflowRouter(tags=["Job"], prefix="/jobs")
+
+
+@job_router.get(
+    "/",
+    responses=create_openapi_http_exception_doc(
+        [status.HTTP_400_BAD_REQUEST, status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN]
+    ),
+)
+def get_jobs(
+    start_date_range: Annotated[
+        RangeFilter,
+        Depends(datetime_range_filter_factory("start_date", Job)),
+    ],
+    end_date_range: Annotated[
+        RangeFilter,
+        Depends(datetime_range_filter_factory("end_date", Job)),
+    ],
+    limit: QueryLimit,
+    offset: QueryOffset,
+    order_by: Annotated[
+        SortParam,
+        Depends(
+            SortParam(
+                [
+                    "id",
+                    "dag_id",
+                    "state",
+                    "job_type",
+                    "start_date",
+                    "end_date",
+                    "latest_heartbeat",
+                    "executor_class",
+                    "hostname",
+                    "unixname",
+                ],
+                Job,
+            ).dynamic_depends(default="id")
+        ),
+    ],
+    session: Annotated[Session, Depends(get_session)],
+    state: str | None = None,
+    job_type: str | None = None,
+    hostname: str | None = None,
+    executor_class: str | None = None,
+    is_alive: bool | None = None,
+) -> JobCollectionResponse:
+    """Get all jobs."""
+    base_select = select(Job).where(Job.state == JobState.RUNNING).order_by(Job.latest_heartbeat.desc())
+    # TODO: Refactor using the `FilterParam` class in commit `574b72e41cc5ed175a2bbf4356522589b836bb11`
+    if state:
+        base_select = base_select.where(Job.state == state)
+    if job_type:
+        base_select = base_select.where(Job.job_type == job_type)
+    if hostname:
+        base_select = base_select.where(Job.hostname == hostname)
+    if executor_class:
+        base_select = base_select.where(Job.executor_class == executor_class)
```

Comment:
Hi @pierrejeambrun,

I'm currently working on `FilterParam` as we discussed in https://github.com/apache/airflow/pull/43407#pullrequestreview-2405200579. The intended usage would be something like `Annotated[ParamFilter[int], Depends(filter_param_factory(filter_name))]`, and the implementation will resemble the structure here: https://github.com/apache/airflow/blob/574b72e41cc5ed175a2bbf4356522589b836bb11/airflow/api_fastapi/common/parameters.py#L222-L268 but will incorporate `FilterOptionEnum`. I believe this could also be refactored within the current WIP.

Reply:
If we want to merge this before the parameter refactoring is finalized then we need to make this PR consistent and self sufficient.

In case the refactoring PR is delayed, not done, or we change our mind about it, code in main should be clean. This is why I recommend extending the current parameters that we have and use them so we can get this merged.

Then we can change all that with the PR improving the parameter system.